HealthcareSystem/Backend/Service/ActionBenefitService/RabbitMqActionBenefitService/RabbitMqActionBenefitService.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Backend.Model.Pharmacies;
+using Backend.Service.Pharmacies;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Backend.Service
+{
+    public class RabbitMqActionBenefitMessageingService : BackgroundService
+    {
+        private IModel _channel;
+        private IConnection _connection;
+        private readonly IServiceProvider _service;
+        private readonly String _hostname;
+        private readonly String _username;
+        private readonly String _password;
+        private readonly String _queueName;
+
+        public RabbitMqActionBenefitMessageingService(IServiceProvider service, IOptions<RabbitMqConfiguration> rabbitMqOPtions)

```

Comment:
I feel like injecting IPharmacyService and IActionsBenefitsService instead of IServiceProvider to make the dependencies more explicit might be better. Also, does the RabbitMQConfiguration need to be added to appsettings.json and read at Startup to be used like this? I don't see a section for it in the appsettings right now, I feel like that will cause issues (like maybe this being null when injected)?

Reply:
Injection must be done like this beacause of different lifetimes of RabbitMqActionBenefitMessageingService(Singleton) and Services(Scope)
For configuration see last commit